ospf6d/ospf6_message.c

```diff
@@ -2356,7 +2356,22 @@ static void ospf6_send_lsupdate(struct ospf6_neighbor *on,
 	}
 	if (oi) {
 		ospf6_packet_add(oi, op);
-		OSPF6_MESSAGE_WRITE_ON(oi);
+		/* If ospf instance is being deleted, send the packet
+		 * immediately
+		 */
+		if ((oi->area == NULL) || (oi->area->ospf6 == NULL))
+			return;
+		if (oi->area->ospf6->inst_shutdown) {
+			struct thread os_packet_thd;
+
+			os_packet_thd.arg = (void *)oi->area->ospf6;
```

Comment:
Nit:
You don't need to cast to void.

Reply:
removed this code as per the below comments